src/app/features/home/home.page.ts

```diff
@@ -377,10 +380,16 @@ export class HomePage {
   }
 
   async navigateToInboxTab() {
+    const hasLoggedIn = await this.diaBackendAuthService.hasLoggedIn();
+    if (!hasLoggedIn) return this.redirectToLoginPage();
     await this.userGuideService.showUserGuidesOnInboxTab();
     await this.userGuideService.setHasOpenedInboxTab(true);
   }
 
+  private async redirectToLoginPage() {
+    this.router.navigate(['/', 'login']);
+  }
+
   logout() {
     const action$ = defer(() => this.mediaStore.clear()).pipe(
       concatMapTo(defer(() => this.database.clear())),
```

Comment:
**ChatGPT response for git diff 4b49d31f.. 8677a8b0 "*home.page.ts"**

Here are some improvement suggestions for the code:

-   Consider extracting the `async ionViewDidEnter()` logic into a separate method that returns an observable, and calling it from `ionViewDidEnter`
-   Extract the code for checking if the user is logged in into a separate method and reuse it where necessary
-   Remove the `async` keyword from the `captureWithCustomCamera` method, it is not necessary as it only calls other functions which are not async
-   Replace the `of(this.onboardingService.isNewLogin)` in `ionViewDidEnter` method with `of(await this.onboardingService.isNewLogin)` so that the value of `isNewLogin` is evaluated before it is passed to the `migrate$` method
-   Add a default case in the `switch` statement in the `presentCaptureActions$` method to handle unexpected values of `this.selectedTabIndex`
-   Use a `switchMap` operator instead of a `switchTapTo` operator in the `ionViewDidEnter` method, as `switchMap` would prevent the creation of multiple subscriptions when navigating to the same component multiple times.
-   Add an explicit type for the error parameter in the catchError operator to provide more clarity.

Reply:
@tammyyang regarding improvements:
- Opinionated 
- Opinionated
- Still need async
- Existing solution is working as expected I tried before rxjs way and restore capture dialog was not showing so lets leave it as it is (aka working)
- `switchTapTo` is working as expected all this time so no need to change
- Overkill, errors can come from different places `migrate`, `promptAppUpdateIfAny`, `onboardingRedirect`, `showUserGuidesOnHomePage` so using generic type `unknow` make sense.